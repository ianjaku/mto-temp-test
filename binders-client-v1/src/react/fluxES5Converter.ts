/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
