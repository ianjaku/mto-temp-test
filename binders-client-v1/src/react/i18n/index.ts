import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
