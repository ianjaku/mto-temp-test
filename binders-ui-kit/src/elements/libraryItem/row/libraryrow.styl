@require "../../../vars.styl"

.library-row
    box-sizing: border-box
    border-top: 1px solid borderGrayColor
    border-left: 1px solid borderGrayColor
    border-right: 1px solid borderGrayColor
    margin-top: -1px
    display: flex
    font-family: defaultFontName
    position: relative
    z-index: 1
    background: whiteColor
    height: 88px
    &.is-draggingAttention
        animation: orb 1.8s linear 1s 1
    &-isReadonly
        color: borderGrayColor
        &.isDocument
            cursor: auto !important
    .statistics
        display: flex
        flex-direction: column
        align-items: center
        text-align: center
        justify-content: center
        color: rgba(29, 29, 29, 0.5)
        float: left
        min-width: 32px
        min-height: 45px
        @media rwd.notmobile
            width: 80px
        margin-left: auto
        .amount
            font-size: 20px
            @media rwd.mobile
                font-size: 14px
        h4
            margin: 0
            font-size: 12px
            font-weight: 100
    &:after
        content: "";
        height: 1px
        width: 100%
        background-color: borderGrayColor
        display: block
        position: absolute
        bottom: 0
        left: 0
    &.is-active:hover
        z-index: 2
        border-color accentColor
    &.is-active:after,:hover:after
        background-color: accentColor
    &:not(.is-locked)
        cursor: pointer
    &-image
        border-right: 1px solid borderGrayColor
    &-content
        word-break: break-word
        width: 100%
        display: flex
        align-items: center
        overflow: hidden
        position: relative
        &-title
            margin-left: 15px
            max-width: 77%
            @media rwd.mobile
                margin-left: 5px
            &-wrapper
                display: flex
                align-items: center
                gap: 8px
                @media rwd.mobile
                    margin-left: 12px
            &-label
                align-self: baseline
                font-size: 16px
                word-wrap: break-word
                pointer-events: none
                .search-hit
                    background-color: accentColor
                    padding: 2px 4px
            &-info
                color: disabledColor
                font-size: 12px
                margin-top: 5px
                padding-bottom: 5px
                display: block
                position: absolute
                bottom: 0px
                height: (1px * additionalInfoHeightDesktop)
                visibility: visible
                background: white
                width: 125% // Force additional info into one row: 100% + (100% - row-title max-width)
                @media rwd.mobile
                    display: none
                &--mobile
                    display: none
                    @media rwd.mobile
                        display: block
                    &:before
                        content: ''
                        display: block
                        position: absolute
                        background: linear-gradient(180deg, rgba(255,255,255,0) 0%, rgba(255,255,255,1) 100%); 
                        bottom: 100%
                        width: 100%
                        left: 0
                        height: 15px
            &-icon
                @media rwd.mobile
                    margin-left: 5px
        &-clickLoader
            margin-left: 15px
        .locked-by
            margin-left: 8px
            display: flex
            &-usericon
                margin-right: 6px
                img
                    width: 15px
            &-dots
                display: flex
                align-items: center
                &-dot
                    display:inline-block
                    width:4px
                    height:4px
                    border-radius:50%
                    margin-right:3px
                    background:middleGreyColor
                    animation: wave 1.3s linear infinite
                    &:nth-child(2)
                        animation-delay: -1.1s
                    &:nth-child(3)
                        animation-delay: -0.9s
        &-right
            margin-left: auto
            display: flex
            align-items: center
        &-icon
            color: darkGreyColor !important
            width: 12px
            &-baseline
                align-self: center
                &.MuiIcon-root
                    font-size: 14px
        .context-menu
            &.is-document
                margin-left: initial
    &--is-pending
        opacity: 0.7
        animation-duration: 1s
        animation-fill-mode: forwards
        animation-iteration-count: infinite
        animation-name: placeHolderShimmer
        animation-timing-function: linear
        background: whiteColor
        background: linear-gradient(to right, whiteColor 10%, lighterGreyColor 28%, whiteColor 43%)
        background-size: 800px 104px
        .library-row
            &-image
                width: 87px
                flex-grow: 0
                flex-shrink: 0
                background-color: lightGreyColor
        &.is-active, &:hover
            border-color: borderGrayColor
        &.is-active:after, &:hover:after
            background-color: borderGrayColor

// MT-2275 Vertical misalignment of title - iPhone
@media rwd.mobile
    .library-row-content-title
        margin-top: 0px
        max-height: 100%
    .library-row--has-progress-bar
        .library-row-content-title
            margin-top: 17px

@keyframes wave
	0%, 60%, 100%
		transform: initial
	30%
		transform: translateY(-10px)


@keyframes placeHolderShimmer
    0%
        background-position: -468px 0
    100%
        background-position: 468px 0

@keyframes orb
    0%
        box-shadow: 0 0 10px -10px accentColor
        z-index: 2
    50%
        box-shadow: 0 0 10px 5px accentColor
        z-index: 2
    100%
        box-shadow: 0 0 10px -10px accentColor
        z-index: 1
