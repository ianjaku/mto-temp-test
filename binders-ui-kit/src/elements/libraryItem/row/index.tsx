import * as React from "react";
import Thumbnail, { FitBehaviour } from "../../thumbnail";
import Tooltip, { TooltipPosition, hideTooltip, showTooltip } from "../../tooltip/Tooltip";
import ClientThumbnail from "@binders/client/lib/clients/repositoryservice/v3/Thumbnail";
import { IChecklistProgress } from "@binders/client/lib/clients/repositoryservice/v3/contract";
import Icon from "../../icons";
import MoreVert from "../../icons/MoreVert";
import ProgressBar from "../../progress-bar";
import { TranslationKeys } from "@binders/client/lib/react/i18n/translations";
import circularProgress from "../../circularprogress";
import { closestByClassName } from "../../../helpers/helpers";
import { contextMenuPopoverClass } from "../../contextmenu";
import cx from "classnames";
import { isMobileView } from "../../../helpers/rwd";
import { menuItemOptionClass } from "../../contextmenu/MenuItem";
import vars from "../../../variables";
import { withTranslation } from "@binders/client/lib/react/i18n";
import "./libraryrow.styl";

// eslint-disable-next-line @typescript-eslint/no-var-requires
const UserIcon = require("../../../../public/icons/user-avatar.svg");

export enum ItemType {
    DOCUMENT = 0,
    COLLECTION = 1,
}

export interface IBasicUserInfo {
    id: string;
    login: string;
    displayName: string;
}

export interface ILibraryRowProps {
    title?: string;
    htmlTitle?: { __html: string };
    thumbnail: ClientThumbnail;
    type?: ItemType;
    libraryRowActions?: ILibraryRowAction[];
    onClick?: () => "pending" | void;
    width?: string;
    className?: string;
    views?: number | JSX.Element;
    lockedBy?: IBasicUserInfo;
    isHidden?: boolean;
    contextMenu?: JSX.Element;
    isPending?: boolean;
    additionalInfo?: string;
    additionalInfoMobile?: string;
    isPublicInfo?: { isPublic?: boolean, isAncestorPublic?: boolean }
    isPublicIconDisabled?: boolean;
    checklistProgress?: IChecklistProgress;
    usePlaceholderThumbnail?: boolean;
    wantsAttention?: boolean;
    shareButton?: React.ReactNode;
}

export interface ILibraryRowState {
    isActive: boolean;
    checklistProgressPercentage?: number;
    isPendingClick?: boolean;
}

export interface ILibraryRowAction {
    title: string;
    iconName: string;
    onClick: () => void;
    className?: string;
}

const iconsStyle = {
    fontSize: "20px",
};

class LibraryRow extends React.Component<ILibraryRowProps, ILibraryRowState> {

    private lockedByTooltip: Tooltip;
    private publicIconTooltip: Tooltip;
    private t;
    private rowContent;
    private rowTitle;

    constructor(props) {
        super(props);
        this.t = props.t;
        this.setActive = this.setActive.bind(this);
        this.onClick = this.onClick.bind(this);
        this.getPublicIcon = this.getPublicIcon.bind(this);
        // this.renderContextMenu = this.renderContextMenu.bind(this);
        this.renderThumbnail = this.renderThumbnail.bind(this);
        this.renderViews = this.renderViews.bind(this);
        this.hideTooltip = this.hideTooltip.bind(this);
        this.showTooltip = this.showTooltip.bind(this);
        this.showPublicTooltip = this.showPublicTooltip.bind(this);
        this.hidePublicTooltip = this.hidePublicTooltip.bind(this);
        this.renderAdditionalInfo = this.renderAdditionalInfo.bind(this);
        this.handleClick = this.handleClick.bind(this);
        this.state = {
            isActive: false,
        };
    }

    public componentDidMount() {
        const { checklistProgress, wantsAttention } = this.props;
        if (checklistProgress) {
            this.calcChecklistProgressPercentage(checklistProgress);
        }
        if (wantsAttention && this.rowContent != null) {
            setTimeout(() => this.rowContent && this.rowContent.scrollIntoView({
                behavior: "smooth",
                block: "start",
            }), 1000);
            window.history.pushState(null, "Manual.to Editor", window.location.origin + window.location.pathname);
        }
    }

    public componentDidUpdate(prevProps) {
        const { checklistProgress } = this.props;
        const { checklistProgress: prevChecklistProgress } = prevProps;
        if (checklistProgress !== prevChecklistProgress) {
            this.calcChecklistProgressPercentage(checklistProgress);
        }
    }

    private calcChecklistProgressPercentage(checklistProgress: IChecklistProgress) {
        const { performed, total } = checklistProgress;
        if (performed === undefined || total === undefined) {
            return;
        }
        this.setState({
            checklistProgressPercentage: performed / total,
        })
    }

    public hideTooltip(e: React.MouseEvent<HTMLElement>): void {
        hideTooltip(e, this.lockedByTooltip);
    }

    public showTooltip(position: TooltipPosition): (e: React.MouseEvent<HTMLElement>) => void {
        return e => {
            showTooltip(e, this.lockedByTooltip, position);
        };
    }

    public renderPending() {
        return (
            <div className={cx("library-row", "library-row--is-pending")}>
                <div className="library-row-image">
                    {this.renderThumbnail()}
                </div>
                <div className="library-row-content">
                    <div className="library-row-content-right">
                        {}
                    </div>
                </div>
            </div>
        );
    }

    public renderChecklistProgress() {
        const { checklistProgressPercentage } = this.state;
        const { checklistProgress } = this.props;
        return checklistProgressPercentage === undefined ?
            (
                <div className="progress-bar" />
            ) :
            (
                <ProgressBar percentage={checklistProgressPercentage} lastUpdated={checklistProgress.lastUpdated} />
            );
    }

    renderShareButton() {
        if (this.props.isPending || !this.props.shareButton) return null;
        return this.props.shareButton;
    }

    public render() {
        const { width, className, lockedBy, isPending, wantsAttention } = this.props;
        const { isActive } = this.state;
        const style = width ? { width } : {};

        return (
            <div>
                <div
                    key="row-default"
                    className={
                        cx(
                            "library-row",
                            { "is-draggingAttention": wantsAttention },
                            { "is-active": isActive },
                            { "is-locked": !!lockedBy },
                            { "library-row--is-pending": isPending },
                            className,
                        )
                    }
                    style={style}
                    onClick={this.onClick}
                >
                    <div className="library-row-image">
                        {!isPending && this.renderThumbnail()}
                    </div>
                    <div className="library-row-content" ref={r => this.rowContent = r}>
                        <div className="library-row-content-title" ref={r => this.rowTitle = r}>
                            {this.renderChecklistProgress()}
                            {this.renderTitle()}
                            {this.renderAdditionalInfo()}
                        </div>
                        {this.renderLockedBy()}
                        <div className="library-row-content-right">
                            {this.renderViews()}
                            {this.renderVisibilityIndicator()}
                            {this.renderShareButton()}
                            {this.renderContextMenu()}
                        </div>
                    </div>
                </div>
            </div>
        );
    }

    private showPublicTooltip(e) {
        showTooltip(e, this.publicIconTooltip, TooltipPosition.TOP);
    }

    private hidePublicTooltip(e) {
        hideTooltip(e, this.publicIconTooltip);
    }

    private maybeRenderPendingClickLoader() {
        if (!this.state.isPendingClick) {
            return null;
        }
        return circularProgress("library-row-content-clickLoader");
    }

    private renderTitle() {
        const { title, htmlTitle, isPublicInfo, isPublicIconDisabled } = this.props;
        const { checklistProgressPercentage } = this.state;
        const publicIcon = isPublicIconDisabled ? { icon: "", message: "" } : this.getPublicIcon(isPublicInfo);
        const labelClassName = "library-row-content-title-label";
        const isCheckListProgressVisible = (checklistProgressPercentage === undefined) ? 0 : 1;
        const areRefsReady = this.rowContent && this.rowTitle;
        const additionalInfoHeight = isMobileView() ? 0 : vars.additionalInfoHeightDesktop;
        const checklistProgressHeight = isMobileView() ? vars.checklistProgressHeightMobile : vars.checklistProgressHeightDesktop;
        const isTitleTooHigh = areRefsReady ?
            (this.rowContent.offsetHeight - (isCheckListProgressVisible * checklistProgressHeight) - additionalInfoHeight < this.rowTitle.offsetHeight) :
            false;

        let marginTop;
        if (isMobileView()) {
            marginTop = isCheckListProgressVisible ? `${checklistProgressHeight}px` : "0";
        } else {
            marginTop = isCheckListProgressVisible ? `${checklistProgressHeight}px` : "5px";
        }

        const styleForTooHighTitle = {
            marginTop,
            alignItems: "flex-start",
        };

        let titleElement;
        if (title) {
            titleElement = <label className={labelClassName}>{title}</label>;
        } else if (htmlTitle) {
            titleElement = <label className={labelClassName} dangerouslySetInnerHTML={htmlTitle} />;
        } else {
            titleElement = <div />;
        }
        const hasCollectionIcon = this.props.type === ItemType.COLLECTION;
        return (
            <div className="library-row-content-title-wrapper" style={isTitleTooHigh ? styleForTooHighTitle : {}}>
                <Tooltip ref={ref => { this.publicIconTooltip = ref; }} message={publicIcon.message} />
                {hasCollectionIcon && (
                    <Icon
                        name="folder"
                        className={cx(
                            "library-row-content-icon library-row-content-icon-baseline",
                        )}
                    />
                )}
                {titleElement}
                {
                    isPublicInfo && publicIcon.icon
                }
                {this.maybeRenderPendingClickLoader()}
            </div>
        );
    }

    private renderAdditionalInfo() {
        const { additionalInfo, additionalInfoMobile, isPending } = this.props;
        if (isPending) {
            return null;
        }
        return (
            <>
                {additionalInfo && <div className="library-row-content-title-info">{additionalInfo}</div>}
                {additionalInfoMobile && <div className="library-row-content-title-info library-row-content-title-info--mobile">{additionalInfoMobile}</div>}
            </>
        );
    }

    private renderVisibilityIndicator() {
        const { isHidden } = this.props;
        return !!isHidden && (
            <Icon name="visibility_off" className={cx("material-icons", "library-row-content-title-icon")} />
        );
    }

    private renderLockedBy() {
        const { lockedBy } = this.props;
        const displayValue = lockedBy && (
            lockedBy.displayName || lockedBy.login || lockedBy.id
        );
        return !!lockedBy && (
            <div
                className="locked-by"
                onMouseEnter={this.showTooltip(TooltipPosition.BOTTOM)}
                onMouseLeave={this.hideTooltip}
            >
                <div className="locked-by-usericon">
                    <img src={UserIcon} />
                </div>
                <div className="locked-by-dots">
                    <span className="locked-by-dots-dot" />
                    <span className="locked-by-dots-dot" />
                    <span className="locked-by-dots-dot" />
                </div>
                <Tooltip ref={ref => { this.lockedByTooltip = ref; }} message={this.t(TranslationKeys.Edit_ItemEditedBy, { displayValue })} />
            </div>
        );
    }

    private renderViews() {
        const { views } = this.props;
        return !views ?
            <div className="statistics" /> :
            (
                <div className="statistics">
                    <h4>{this.t(TranslationKeys.Analytics_View, { count: 2 })}</h4>
                    <span className="amount">
                        {views}
                    </span>
                </div>
            );
    }

    private renderContextMenu() {
        const { isPending, contextMenu } = this.props;
        return (isPending) ?
            MoreVert({ marginRight: "12px", color: vars.middleGreyColor }) :
            contextMenu;
    }

    private renderThumbnail() {
        const { thumbnail } = this.props;
        const thumbnailSrc = thumbnail.buildRenderUrl ?
            thumbnail.buildRenderUrl() :
            thumbnail.medium;

        return (
            <Thumbnail
                usePlaceholder={this.props.usePlaceholderThumbnail}
                fitBehaviour={thumbnail.fitBehaviour === "crop" ? FitBehaviour.CROP : FitBehaviour.FIT}
                bgColor={thumbnail.bgColor}
                src={thumbnailSrc}
                width={86}
                stretchImage={false}
                rotation={thumbnail.rotation ? parseInt(thumbnail.rotation, 10) : 0}
            />
        );
    }

    private getPublicIcon(publicInfo) {
        const { type } = this.props;

        if (!publicInfo) {
            return {};
        }
        const { isPublic, hasPublicAncestors } = publicInfo;
        if (!isPublic && !hasPublicAncestors) {
            return {};
        }

        return {
            icon: (
                <Icon
                    onMouseEnter={this.showPublicTooltip}
                    onMouseLeave={this.hidePublicTooltip}
                    themeColor={hasPublicAncestors ? "disabled" : "primary"}
                    style={iconsStyle}
                    className={cx("material-icons", "library-row-content-title-icon")}
                    name="public"
                />
            ),
            message: hasPublicAncestors ?
                this.t(TranslationKeys.DocManagement_ItemInPublic, { type: ItemType[type].toLowerCase() }) :
                this.t(TranslationKeys.DocManagement_ItemPublic, { type: ItemType[type].toLowerCase() }),
        };
    }

    private handleClick(attempt = 0) {
        if (attempt > 100) {
            this.setState({ isPendingClick: false });
            return;
        }
        const { onClick } = this.props;
        const clickResult = onClick();
        if (clickResult === "pending") {
            if (attempt === 0) {
                this.setState({ isPendingClick: true });
            }
            setTimeout(() => this.handleClick(attempt + 1), 200);
            return;
        }
        this.setState({ isPendingClick: false });
    }

    private onClick({ target }) {
        // IE vs the other browsers (the saga continues)
        //
        // Most browsers fill in the className field
        // IE does not, but does give us the target.type
        const { lockedBy, onClick } = this.props;

        const expandMenuClicked = (
            target.type === "button" ||
            target.className.includes("contextMenu-icon") ||
            target.className.includes(menuItemOptionClass) ||
            closestByClassName(target, "modal")
        );
        if (target.parentElement && target.parentElement.className.includes(contextMenuPopoverClass)) {
            return;
        }
        if (!expandMenuClicked && onClick && !lockedBy && !this.state.isPendingClick) {
            this.handleClick();
        }
    }

    private setActive(isActive) {
        this.setState({
            isActive,
        });
    }
}

export default withTranslation()(LibraryRow);
