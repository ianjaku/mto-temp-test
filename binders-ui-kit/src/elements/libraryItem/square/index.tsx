import * as React from "react";
import Thumbnail, { FitBehaviour } from "../../thumbnail";
import ContextMenu from "../../contextmenu/";
import Icon from "../../icons";
import { ItemType } from "./../row/index";
import MenuItem from "../../contextmenu/MenuItem";
import classNames from "classnames";
import "./librarysquare.styl";

export interface ILibraryItemProps {
    fitBehaviour?: FitBehaviour;
    title: string;
    thumbnail: string;
    type?: ItemType;
    librarySquareActions?: ILibraryItemAction[];
    onClick?: () => void;
    width?: string;
}

export interface ILibrarySquareState {
    isActive: boolean;
}

export interface ILibraryItemAction {
    title: string;
    iconName: string;
    onClick: () => void;
}

const iconStyles = {
    color: "#cdccca",
    fontSize: "20px",
};

const menuStyles = {
    height: "auto",
    padding: "2px 5px",
    width: "auto",
};

const menuIconStyles = {
    fontSize: "20px",
};

class LibrarySquare extends React.Component<ILibraryItemProps, ILibrarySquareState> {

    private type: ItemType;
    private icon;

    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
    constructor(props) {
        super(props);
        this.type = this.setItemType(props.type);
        this.icon = this.getIconType(this.type);
        this.setActive = this.setActive.bind(this);
        this.onClick = this.onClick.bind(this);
        this.state = {
            isActive: false,
        };
    }

    public render(): JSX.Element {

        const { title, thumbnail, librarySquareActions, fitBehaviour = FitBehaviour.CROP } = this.props;
        const { isActive } = this.state;
        const itemKindClass = this.setItemClass(this.type);
        const contextMenu = librarySquareActions && librarySquareActions.length > 0 ?
            <ContextMenu menuIconName={"more_vert"} onChangeOpened={this.setActive}  menuStyle={menuStyles} menuIconStyle={menuIconStyles}>
                {librarySquareActions.map(librarySquareAction => <MenuItem key={librarySquareAction.title} {...librarySquareAction} />)}
            </ContextMenu> :
            undefined;
        return (
            <div className={classNames("libraryItem", itemKindClass, {"is-active": isActive })} onClick={this.onClick} >
                <Thumbnail fitBehaviour={fitBehaviour} src={thumbnail} title={title} width={170} />
                <div className="libraryItem-content">
                    <div className="libraryItem-info">
                        <span className="libraryItem-title">{title}</span>
                        <span className="libraryItem-icon">{this.icon}</span>
                    </div>
                    <div className="libraryItem-menu">
                        {contextMenu}
                    </div>
                </div>
            </div>
        );
    }

    private setItemType(type: ItemType) {
        return type;
    }

    private getIconType(type: ItemType) {
        let iconName;
        switch (type) {
            case ItemType.COLLECTION:
                iconName = "folder_open";
                break;
            default:
                iconName = "";
        }
        return (<Icon style={iconStyles} name={iconName} />);
    }

    private setActive(isActive) {
        this.setState({
            isActive,
        });
    }

    private setItemClass(type: ItemType) {
        switch (type) {
            case ItemType.COLLECTION:
                return "collection";
            case ItemType.DOCUMENT:
                return "document";
            default:
                return "";
        }
    }

    private onClick({ target }) {
        if (!target.className.includes("contextMenu-icon") && this.props.onClick) {
            this.props.onClick();
        }
    }
}

export default LibrarySquare;
