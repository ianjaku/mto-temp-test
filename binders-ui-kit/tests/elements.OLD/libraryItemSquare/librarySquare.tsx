import * as React from "react";
import LibrarySquare, { ILibraryItemAction } from "../../../src/elements/libraryItem/square";
import { ItemType } from "../../../src/elements/libraryItem/row";
import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
import { create } from "react-test-renderer";
import { mount } from "enzyme";

const actions: ILibraryItemAction[] = [
    {
        iconName: "edit",
        onClick: undefined,
        title: "action 1",
    },
    {
        iconName: "delete",
        onClick: undefined,
        title: "action 2",
    },
];

function wrapWithTheme(node) {
    return (
        <MuiThemeProvider>{node}</MuiThemeProvider>
    );
}

const createCollectionItem = () => (
    wrapWithTheme(<LibrarySquare title="Universitet Gent" thumbnail="https://dummyimage.com/400x400/cdccca" type={ItemType.COLLECTION} librarySquareActions={actions} />)
);

const createDocumentItem = () => (
    wrapWithTheme(<LibrarySquare title="Universitet Gent" thumbnail="https://dummyimage.com/400x400/cdccca" type={ItemType.DOCUMENT} librarySquareActions={actions} />)
);

describe("My Square Library", () => {

    test("Base Library with single Collection (snapshot)", () => {
        const item = create(createCollectionItem());
        const serialized = item.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Base Library with single Document (snapshot)", () => {
        const item = create(createDocumentItem());
        const serialized = item.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Collection - clicking contextMenu icon (mount: enzyme+jsdom)", () => {
        const item = mount(createCollectionItem());
        item.find(".libraryItem-menu button").first()
            .simulate("click");
        expect(item.find("button").hasClass("contextMenu-icon--active"));
    });

    test("Document - clicking contextMenu icon (mount: enzyme+jsdom)", () => {
        const item = mount(createDocumentItem());
        item.find(".libraryItem-menu button").first()
            .simulate("click");
        expect(item.find("button").hasClass("contextMenu-icon--active"));
    });

    test("Item with correctly (kind) class name (mount: enzyme+jsdom)", () => {
        const item = mount(createCollectionItem());
        const isCollectionClass = item.find(".libraryItem").first();
        expect(isCollectionClass.hasClass("collection")).toEqual(true);
    });

    test("Item with correctly (kind) class name (mount: enzyme+jsdom)", () => {
        const item = mount(createDocumentItem());
        const isDocumentClass = item.find(".libraryItem").first();
        expect(isDocumentClass.hasClass("document")).toEqual(true);
    });

});
