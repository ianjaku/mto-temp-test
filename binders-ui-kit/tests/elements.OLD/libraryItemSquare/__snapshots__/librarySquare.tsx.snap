// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Square Library Base Library with single Collection (snapshot) 1`] = `
<div
  className="libraryItem collection"
  onClick={[Function]}
>
  <div
    className="thumbnail-outer-wrapper"
    style={
      Object {
        "backgroundColor": "##ffffff",
        "height": 170,
        "width": 170,
      }
    }
  >
    <div
      className="thumbnail-wrapper crop landscape"
      onClick={undefined}
      onDoubleClick={undefined}
      onMouseDownCapture={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "##ffffff",
          "height": 170,
          "width": 170,
        }
      }
    >
      <img
        className="thumbnail crop landscape"
        onLoad={[Function]}
        src="https://dummyimage.com/400x400/cdccca"
        style={
          Object {
            "height": 170,
            "left": "50%",
            "position": "absolute",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "auto",
          }
        }
        title="Universitet Gent"
      />
    </div>
  </div>
  <div
    className="libraryItem-content"
  >
    <div
      className="libraryItem-info"
    >
      <span
        className="libraryItem-title"
      >
        Universitet Gent
      </span>
      <span
        className="libraryItem-icon"
      >
        <span
          className="material-icons"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#cdccca",
              "display": "inline-block",
              "fontSize": "20px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          folder_open
        </span>
      </span>
    </div>
    <div
      className="libraryItem-menu"
    >
      <div
        className=""
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "position": "relative",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 0,
              "fontWeight": "inherit",
              "height": "auto",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "padding": "2px 5px",
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "auto",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <span
              className="contextMenu-icon material-icons"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fontSize": "20px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                }
              }
            >
              more_vert
            </span>
          </div>
        </button>
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`My Square Library Base Library with single Document (snapshot) 1`] = `
<div
  className="libraryItem document"
  onClick={[Function]}
>
  <div
    className="thumbnail-outer-wrapper"
    style={
      Object {
        "backgroundColor": "##ffffff",
        "height": 170,
        "width": 170,
      }
    }
  >
    <div
      className="thumbnail-wrapper crop landscape"
      onClick={undefined}
      onDoubleClick={undefined}
      onMouseDownCapture={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "##ffffff",
          "height": 170,
          "width": 170,
        }
      }
    >
      <img
        className="thumbnail crop landscape"
        onLoad={[Function]}
        src="https://dummyimage.com/400x400/cdccca"
        style={
          Object {
            "height": 170,
            "left": "50%",
            "position": "absolute",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "auto",
          }
        }
        title="Universitet Gent"
      />
    </div>
  </div>
  <div
    className="libraryItem-content"
  >
    <div
      className="libraryItem-info"
    >
      <span
        className="libraryItem-title"
      >
        Universitet Gent
      </span>
      <span
        className="libraryItem-icon"
      >
        <span
          className="material-icons"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#cdccca",
              "display": "inline-block",
              "fontSize": "20px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          
        </span>
      </span>
    </div>
    <div
      className="libraryItem-menu"
    >
      <div
        className=""
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "position": "relative",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 0,
              "fontWeight": "inherit",
              "height": "auto",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "padding": "2px 5px",
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "auto",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <span
              className="contextMenu-icon material-icons"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fontSize": "20px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                }
              }
            >
              more_vert
            </span>
          </div>
        </button>
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
